chia/consensus/multiprocess_validation.py

```diff
@@ -220,13 +223,19 @@ async def pre_validate_blocks_multiprocessing(
             recent_blocks[curr.header_hash] = curr
             if curr.is_transaction_block:
                 num_blocks_seen += 1
+            if len(recent_plot_ids) < 4:
+                # TODO: Add `get_proof_of_space()` to BlockStore.
+                full_block = await block_store.get_full_block(curr.header_hash)
```

Comment:
this lookup has the potential of being quite expensive. I think it would make sense to benchmark syncing with this patch

Reply:
Thinking to extract here just the `ProofOfSpace`, not the full block (and add to `BlockStore` a new SQL query that selects only that). But this is still WIP, trying to get all tests pass first.

Reply:
if you only need the plot_id we can create a cache for that instead of going to the disk 4 times on each block validation